src/components/button/button.jsx

```diff
@@ -18,7 +18,7 @@ const Button = (props) => {
   const fixedProps = _.extend(
     {},
     _.omit(props, ['data', 'size', 'purpose']),
-    _.extend({}, props, flatten(props.data)),
```

Comment:
We don't need to pass all of props + data in here as we're already passing them in on the previous line.